ios-base/DroidKaigi 2020/Session/Views/Cells/SessionCell.swift

```diff
@@ -76,8 +76,32 @@ final class SessionCell: UICollectionViewCell {
         let view = UIControl()
         let speakerIconView = UIImageView()
         if let imageURL = imageURL {
-            let options = ImageLoadingOptions(transition: .fadeIn(duration: 0.3))
-            Nuke.loadImage(with: imageURL, options: options, into: speakerIconView)
+            // Configure cache
+            ImageCache.shared.costLimit = 1024 * 1024 * 50 // 50 MB
+            ImageCache.shared.countLimit = 100
+            ImageCache.shared.ttl = 120 // Invalidate image after 120 sec
+
+            let request = ImageRequest(
+                url: imageURL
+            )
+
+            if let cachedImage = ImageCache.shared[request] {
+                speakerIconView.image = cachedImage
+            } else {
+                ImagePipeline.shared.loadImage(with: imageURL) { result in
+                    switch result {
+                    case let .success(response):
+                        ImageCache.shared[request] = response.image // Set cache
+                        UIView.transition(with: speakerIconView,
+                                          duration: 0.3,
+                                          options: .transitionCrossDissolve,
+                                          animations: { speakerIconView.image = ImageCache.shared[request] },
+                                          completion: nil)
```

Comment:
When displaying the cached image, the image can be read immediately, so I thought that fadeIn was unnecessary.
For this reason, fadeIn is only given when reading from the server.